### ISSUE TITLE
Impossible to automate assembler crafting

### ISSUE THREAD

**Issue by walia6** [ValkyrieofNight/EnvironmentalTech#255]
Assembler recipes such as the Redstone Infused Obsidian Plate  or other recipes that use 5 item slots to craft, are impossible to automate with automation mods such as Applied Energistics 2. There are simply not enough faces on a cube to have an input face for every slot, one output face for the input slot, and a face for power.

Not a bug, but an oversight and an issue nonetheless.

**Comment by LTenshi** [comment]
It is possible to use Xnet to put one item in at a time, but I wholeheartedly agree with you. It's not an easy machine to automate and to make matters worse, you can't even speed it up. It becomes a logistical nightmare to add more machines to speed up the crafting process.

**Comment by walia6** [comment]
> 
> 
> It is possible to use Xnet to put one item in at a time, but I wholeheartedly agree with you. It's not an easy machine to automate and to make matters worse, you can't even speed it up. It becomes a logistical nightmare to add more machines to speed up the crafting process.

Good point - not closing as the dev may see this as an accessibility issue.

**Comment by ValkyrieofNight** [comment]
This is not a final product the mod is in alpha I am aware of the inability to automate it and how slow it is, this is obviously not finished.

**Comment by uecasm** [comment]
Since most of the recipes are either center + one edge or center + four surround, it should theoretically be possible to configure one side to center, one side to top, one side to the other three slots, and one side to output, leaving one side for power and one side to look pretty.

Or have two separate assembler setups for the two and five ingredient recipes, which would let you save another side on the five by merging the top slot.

Having said that, thus far I've had absolutely no luck accessing the assembler with any automation mod that I've tried -- regardless of side config, they're all telling me that the assembler is simply not a valid inventory and they refuse to put items of any kind into it.

**Comment by VorpalLemur** [comment]
Came here to comment on this.   This issue broke my interest in exploring this version of the mod.   It's just not workable.   Mind you, I really enjoy automation challenges, they're why I play modded minecraft, and have made some pretty complex builds in pursuit of them.  But this?   

This reminds me of a somewhat similar issue Galacticraft had with their plate-making machine.   In the end, after countless player complaints, the developer had to put a lot of hidden and complex magic code on the input side of a machine so that, when items were added one at a time by a standard hopper (and this was the only time it worked, so your Refined Storage crafter actually had to input to a vanilla hopper so the hopper could slowly drip feed the items into the machine, what a joke), the machine was able to make a semi-intelligent guess as to what you were trying to craft and place the items accordingly.  It only worked most of the time, and machine jams were common.   Ultimately there was one recipe, middle of the progression tier, that simply was not automatable, because the logic wasn't able to distinguish that one particular recipe from a similar one. 

This was in Sevtech: Ages of the Sky where I played this.  Banging my head on this issue over and over again wasn't fun, it didn't feel like a challenge but rather an ordeal.     Ultimately the Sevtech modpack developers used another mod, Modular Machinery, to implement an entirely custom and original multiblock machine in the game to make plates with a simpler automation interface, just in order to get around this flawed design in Galacticraft, presumably because the Galcticraft dev wasn't willing to fix it.

The EnvTech Assembler and its associated recipes are more complex than the plate machine was in Galacticraft.

I hope Env Tech doesn't end up going down the same road and making the same mistakes.   I'm glad to hear that this isn't the final design.  cheers, and thanks for the cool mod

**Comment by quietust** [comment]
This appears to be fixed as of 3.0.7.0, but it's effectively been replaced by #270.

**Comment by MpVrooyeN** [comment]
I'm playing the Direwolf20 1.16 pack and I'm having the same issues. I was able to automate the assembler with Xnet but it'll craft one item with it and then I'll have to break and replace the assembler over and over.

**Comment by IamStig** [comment]
I have managed to get the assembler auto crafting using pretty pipes ....except that refined storage patterns are impossible to get to work (I have done the craft an item and replace the phantom item in the pattern to craft the next tier but it fails).
